src/Fixer/Alias/MbStrFunctionsFixer.php

```diff
@@ -42,6 +43,23 @@ final class MbStrFunctionsFixer extends AbstractFunctionReferenceFixer
         'substr_count' => ['alternativeName' => 'mb_substr_count', 'argumentCount' => [2, 3, 4]],
     ];
 
+    /**
+     * @var array<string, array>
+     */
+    private $functions;
+
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->functions = array_filter(
+            self::$functionsMap,
+            static function (array $mapping) {
+                return \function_exists($mapping['alternativeName']);
```

Comment:
this is work way to do the check. during runtime, we have ext-mb polyfilled, thus we alternativeName always exists

vide https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4380#discussion_r319718704
vide #4529

Reply:
[ReflectionFunctionAbstract::isInternal](https://www.php.net/manual/en/reflectionfunctionabstract.isinternal.php) can help: https://3v4l.org/Wdp27

Reply:
Looking nice!
Yet, to be honest, i'm not sure is it the way we want to follow - implicit handling of decision "should we transform this function or not" by detecting `isInternal` or current runtime (that still can be different than end environment), or explicit configuration (like we do eg for PHPUnit - we aim version X or Y, regardless what is installed in repo we are fixing).
What do you think, @Slamdunk ?

Reply:
I'm gonna talk about this particular Fixer only: this Fixer aims to not forget anywhere any non-multi-byte string comparison. Why should anyone could want to fix one function but not the others?

> by detecting `isInternal` or current runtime (that still can be different than end environment)

That would be a user's fault: this tool it is intended to be run with the same binary used by the production, so no worry to me about this point

> explicit configuration (like we do eg for PHPUnit - we aim version X or Y, regardless what is installed in repo we are fixing)

As a user myself of PHP-CS-Fixer the fact that PHPUnit related fixers are by explicit configuration represents just a lack of the tool, not a cool feature. What would be the purpose of having PHPUnit 999 installed and tells PHP-CS-Fixer to to transition the code from PHPUnit 1 to PHPUnit 2? The installed version and targeted transition one should always be the same.
Here I don't question what was done: it is a safety net for PHP-CS-Fixer maintainers and it's good for now; just talking about theory here :wink:

Reply:
> Why should anyone could want to fix one function but not the others?

vide issue reported - because he does not had the other function (`mb_str_split`) and code execution after fixing was crashing for him (We had installed polyfill, he didn't).

for phpunit - well, one can has it installed as global dependency or phar file, and we cannot autodetect version.

agree - if we expect to run on same runtime (which we do), then it's not an issue if we don't modify the runtime on the fly (by adding pollyfills). then, maybe `isInternal` is good enough for us here.